backend/corpora/common/entities/entity.py

```diff
@@ -59,6 +60,46 @@ def __getattr__(self, name):
         """
         If the attribute is not in Entity, return the attribute in db_object.
         :param name:
-        :return:
         """
         return self.db_object.__getattribute__(name)
+
+    @classmethod
+    def _create_sub_objects(
+        cls, rows: typing.List[dict], db_table: Base, add_columns: dict = None, primary_keys: typing.List[str] = None
+    ) -> typing.List[Base]:
+        """
+        Create or modify `rows` in `db_table` associated with Entity Object during object creation. If id is provided,
+        then the row with that id is retrieved and updated, else a new UUID is generated and a new row is created.
+
+        :param rows: A list of dictionaries each specifying a row to insert or modify
+        :param db_table: The Table to add or modify rows
+        :param primary_keys: Additional columns required to build the primary key. This is used when the primary consist
+        of multiple columns.
+        :param add_columns: Additional columns attributes or modifications to add to the row.
+
+        This can be used when there are shared column values that need to be added across all the new rows.
+        For example: DbProjectLink generated for a specific project should all have the same DbProjectLink.project_id
+        and DbProjectLink.project_status. The function call would be:
+        >>>> cls._create_sub_objects(
+        >>>>    [{'link_url':'abc', 'link_type': ProjectLinkType.OTHER}],
+        >>>>    DbProjectLink,
+        >>>>    add_columns={'project_id':'abcd','project_status':ProjectStatus.EDIT}
+        >>>>    )
+
+        Another use would be to overwrite column specified in the rows.
+
+       :return: a list of database objects to create or modify.
+        """
+        add_columns = add_columns if add_columns else {}
+        db_objs = []
+        for columns in rows:
+            _columns = dict(**columns, **add_columns)  # add_columns changes takes precedence over columns
```

Comment:
What does it mean for add_columns changes to take "precedence" over columns?

Reply:
if there are matching keys in columns and add_columns, the key value in add_columns will be used.

Reply:
Can you update the comment with that please? Your comment above makes it much more clear :)